### PR TITLE
Use secret environment variable values in Kubernetes-based tasks

### PR DESCRIPTION
**Pull request by jwitz**
Resolves https://github.com/astronomer/docs/issues/1930
Resolves https://github.com/astronomer/docs/issues/2025

Software content is still WIP, but Astro content is ready for review

### CODE REVIEW

File: astro/executors.md [hunk 1]
```diff
@@ -180,6 +180,92 @@ with DAG(
...
+1. Add the following import to your DAG file:
+   
+    ```python
+    from airflow.kubernetes.secret import Secret
+    ```
+
+2. Define a Kubernetes `Secret` in your DAG instantiation in the following format:
+
+    ```python
+    secret_env = Secret(deploy_type="env", deploy_target="<SECRET_KEY>", secret="env-secrets", key="<SECRET_KEY>")
```
Comment (RNHTTR) [inline astro/executors.md:198]: I think the nomenclature `<SECRET_KEY>` implies the use of an private key (e.g. the private key in a public/private key pair for API credentials). Could we instead say something like...

> Astro [environment variables](environment-variables.md) marked as secrets are stored in a Kubernetes secret called `env-secrets`. To use a secret value in a task running on the KubernetesExecutor, mount the secret to the Pod running the task. The name of an environment variable (`<ENV_VAR>` in the example below) can be used to create a Kubernetes secret in your DAG in order to access the environment variable's value at runtime.

File: astro/kubernetespodoperator.md [hunk 2]
```diff
@@ -217,6 +217,58 @@ If your Docker image is hosted in an Amazon ECR repository, add a permissions po
     - In the Amazon ECR Dashboard, click **Repositories** in the left menu.
     - Click the **Private** tab and then copy the URI of the repository that hosts the Docker image.
 
+## Use secret environment variables with the KubernetesPodOperator
+
+Astro [environment variables](environment-variables.md) marked as secrets are stored in a Kubernetes secret called `env-secrets`. To use a secret value in a task running the KubernetesPodOperator, mount the secret to the Pod running the task.
```
Comment (RNHTTR) [inline astro/kubernetespodoperator.md:222]: Same as my comment for the k8s executor:

I think the nomenclature `<SECRET_KEY>` implies the use of an private key (e.g. the private key in a public/private key pair for API credentials). Could we instead say something like...

> Astro [environment variables](environment-variables.md) marked as secrets are stored in a Kubernetes secret called `env-secrets`. To use a secret value in a task running on the KubernetesExecutor, mount the secret to the Pod running the task. The name of an environment variable (`<ENV_VAR>` in the example below) can be used to create a Kubernetes secret in your DAG in order to access the environment variable's value at runtime.

File: software/kubernetes-executor.md [hunk 3]
```diff
@@ -5,78 +5,313 @@ id: kubernetes-executor
...
 
-To configure these resources for each Pod, you configure a Pod template. For more information on configuring Pod template values, reference the [Kubernetes documentation](https://kubernetes.io/docs/concepts/workloads/pods/#pod-templates).
+To configure these resources for a given tasks's Pod, you specify a `pod_override` in your DAG code. To specify a Pod template for many or all of your tasks, you can write a helper function to construct a `pod_override` in your DAGs or configure a global For more information on configuring Pod template values, reference the [Kubernetes documentation](https://kubernetes.io/docs/concepts/workloads/pods/#pod-templates).
 
 ## Prerequisites
 
 You must have an Airflow Deployment on Astronomer running with the Kubernetes executor. For more information on configuring an executor, see [Configure a Deployment](configure-deployment.md). To learn more about different executor types, see [Airflow executors explained](https://docs.astronomer.io/learn/airflow-executors-explained).
 
-## Configure the Kubernetes executor using pod templates
+## Configure the default worker Pod for all Deployments
```
Comment (RNHTTR) [inline software/kubernetes-executor.md:22]: I think it'd be helpful to also include an order of precedence for overrides. For example, I believe the helm chart overrides any other resource definitions.

File: astro/environment-variables.md [hunk 4]
```diff
@@ -67,11 +67,14 @@ After you set an environment variable key, only the environment variable value c
 
 Non-secret environment variables set in the Cloud UI are stored in a database that is managed by Astronomer and hosted in the Astro control plane. When you configure a secret environment variable in the Cloud UI, the following methodology is used:
 
-- Astro generates a manifest that defines a Kubernetes secret containing your variable's key and value.
+- Astro generates a manifest that defines a Kubernetes secret, named `env-secrets` containing your variable's key and value.
```
Comment (lzdanski) [inline astro/environment-variables.md:70]:
```suggestion
- Astro generates a manifest that defines a Kubernetes secret, named `env-secrets`, that contains your variable's key and value.
```

File: astro/kubernetespodoperator.md [hunk 5]
```diff
@@ -217,6 +217,58 @@ If your Docker image is hosted in an Amazon ECR repository, add a permissions po
...
+
+1. Add the following import to your DAG file:
+   
+    ```python
+    from airflow.kubernetes.secret import Secret
+    ```
+
+2. Define a Kubernetes `Secret` in your DAG instantiation in the following format:
```
Comment (lzdanski) [inline astro/kubernetespodoperator.md:230]:
```suggestion
2. Define a Kubernetes `Secret` in your DAG instantiation using the following format:
```

File: astro/kubernetespodoperator.md [hunk 6]
```diff
@@ -217,6 +217,58 @@ If your Docker image is hosted in an Amazon ECR repository, add a permissions po
...
+
+    ```python
+    secret_env = Secret(deploy_type="env", deploy_target="<SECRET_KEY>", secret="env-secrets", key="<SECRET_KEY>")
+    namespace = conf.get("kubernetes", "NAMESPACE")
+    ```
+   
+3. Reference the key for the environment variable, formatted as `$VARIABLE_KEY` in the task using the KubernetesPodOperator.
```
Comment (lzdanski) [inline astro/kubernetespodoperator.md:237]:
```suggestion
3. Reference the key for the environment variable, formatted as `$VARIABLE_KEY` in the task using the `KubernetesPodOperator`.
```
Is the code formatting necessary here? I know sometimes with Kube/Docker stuff, the code formatting can vary.

File: software/kubernetes-executor.md [hunk 7]
```diff
@@ -5,78 +5,313 @@ id: kubernetes-executor
 description: Run and configure the Kubernetes executor on Astronomer.
 ---
 
-The Apache Airflow [Kubernetes Executor](https://airflow.apache.org/docs/apache-airflow/stable/executor/kubernetes.html) relies on a fixed single Pod that dynamically delegates work and resources. For each task that needs to run, the executor talks to the Kubernetes API to dynamically launch Pods which terminate when that task is completed.
+The [Kubernetes Executor](https://airflow.apache.org/docs/apache-airflow/stable/executor/kubernetes.html) creates individual Pods that dynamically delegate work and resources to individual tasks. For each task that needs to run, the executor talks to the Kubernetes API and dynamically launches Pods which terminate when the task is completed.
```
Comment (lzdanski) [inline software/kubernetes-executor.md:8]: **the executor talks to the Kubernetes API** - do we have any anthropomorphism or colloquialism style requirements?  > communicates with, interacts with, could be alternative phrasing

File: software/kubernetes-executor.md [hunk 8]
```diff
@@ -5,78 +5,313 @@ id: kubernetes-executor
...
 
-This enables the executor to scale depending on how many Airflow tasks you're running at a given time. It also means you can configure the following for each individual Airflow task:
+You can customize your Kubernetes Pods to scale depending on how many Airflow tasks you're running at a given time. It also means you can configure the following for each individual Airflow task:
 
 - Memory allocation
 - Service accounts
 - Airflow image
 
-To configure these resources for each Pod, you configure a Pod template. For more information on configuring Pod template values, reference the [Kubernetes documentation](https://kubernetes.io/docs/concepts/workloads/pods/#pod-templates).
+To configure these resources for a given tasks's Pod, you specify a `pod_override` in your DAG code. To specify a Pod template for many or all of your tasks, you can write a helper function to construct a `pod_override` in your DAGs or configure a global For more information on configuring Pod template values, reference the [Kubernetes documentation](https://kubernetes.io/docs/concepts/workloads/pods/#pod-templates).
```
Comment (lzdanski) [inline software/kubernetes-executor.md:16]:
```suggestion
To configure these resources for a given tasks's Pod, you specify a `pod_override` in your DAG code. To specify a Pod template for many or all of your tasks, you can write a helper function to construct a `pod_override` in your DAGs or configure a global setting. For more information on configuring Pod template values, reference the [Kubernetes documentation](https://kubernetes.io/docs/concepts/workloads/pods/#pod-templates).
```

Missing a couple words - took a guess.

**Review by lzdanski — COMMENTED** [review]
Did a style/grammar editorial pass. Small stuff, but there's a couple places where it looked like a word might be missing.

File: astro/executors.md [hunk 9]
```diff
@@ -180,6 +180,92 @@ with DAG(
...
+1. Add the following import to your DAG file:
+   
+    ```python
+    from airflow.kubernetes.secret import Secret
+    ```
+
+2. Define a Kubernetes `Secret` in your DAG instantiation using the following format:
+
+    ```python
+    secret_env = Secret(deploy_type="env", deploy_target="<SECRET_KEY>", secret="env-secrets", key="<SECRET_KEY>")
```
Comment (jwitz) [inline astro/executors.md:198]:
```suggestion
    secret_env = Secret(deploy_type="env", deploy_target="<VARIABLE_KEY>", secret="env-secrets", key="<VARIABLE_KEY>")
```

File: astro/executors.md [hunk 10]
```diff
@@ -180,6 +180,92 @@ with DAG(
 
 When this DAG runs, it launches a Kubernetes Pod with exactly 0.5m of CPU and 1024Mi of memory as long as that infrastructure is available in your cluster. Once the task finishes, the Pod terminates gracefully.
 
+### Mount secret environment variables to worker Pods
+
+<!-- Same content in other products -->
+
+Astro [environment variables](environment-variables.md) marked as secrets are stored in a Kubernetes secret called `env-secrets`. To use a secret value in a task running on the KubernetesExecutor, mount the secret to the Pod running the task.
```
Comment (jwitz) [inline astro/executors.md:187]:
```suggestion
Astro [environment variables](environment-variables.md) marked as secrets are stored in a Kubernetes secret called `env-secrets`. To use a secret value in a task running on the KubernetesExecutor, mount the secret to the Pod running the task as a Kubernetes Secret. Your DAG will pull the secret from your Deployment at runtime and apply it to this Secret.
```

File: astro/executors.md (hunk 1 above)
Comment (jwitz) [reply astro/executors.md:198]: I updated this slightly to be more direct about what the code is doing. I didn't realize we were creating a new Secret in the DAG - thought it was just a way of calling the Secret that already existed.

File: astro/kubernetespodoperator.md [hunk 11]
```diff
@@ -217,6 +217,58 @@ If your Docker image is hosted in an Amazon ECR repository, add a permissions po
...
+1. Add the following import to your DAG file:
+   
+    ```python
+    from airflow.kubernetes.secret import Secret
+    ```
+
+2. Define a Kubernetes `Secret` in your DAG instantiation using the following format:
+
+    ```python
+    secret_env = Secret(deploy_type="env", deploy_target="<SECRET_KEY>", secret="env-secrets", key="<SECRET_KEY>")
```
Comment (jwitz) [inline astro/kubernetespodoperator.md:233]:
```suggestion
    secret_env = Secret(deploy_type="env", deploy_target="<VARIABLE_KEY>", secret="env-secrets", key="<VARIABLE_KEY>")
```

File: astro/kubernetespodoperator.md (hunk 2 above)
Comment (jwitz) [inline astro/kubernetespodoperator.md:222]:
```suggestion
Astro [environment variables](environment-variables.md) marked as secrets are stored in a Kubernetes secret called `env-secrets`. To use a secret value in a task running on the KubernetesExecutor, you pull the value from `env-secrets` and mount it to the Pod running your task as a new Kubernetes Secret.
```

File: astro/executors.md (hunk 1 above)
Comment (RNHTTR) [reply astro/executors.md:198]: I'm not sure that I agree that `VARIABLE_KEY` is much better given the concept of Airflow Variables compared to environment variables. Maybe `<ENVIRONMENT_VARIABLE_NAME>`?

File: software/kubernetes-executor.md (hunk 3 above)
Comment (RNHTTR) [reply software/kubernetes-executor.md:22]: I think I had it backwards, and this should be good.

File: software/kubernetes-executor.md [hunk 12]
```diff
@@ -5,78 +5,313 @@ id: kubernetes-executor
...
 You must have an Airflow Deployment on Astronomer running with the Kubernetes executor. For more information on configuring an executor, see [Configure a Deployment](configure-deployment.md). To learn more about different executor types, see [Airflow executors explained](https://docs.astronomer.io/learn/airflow-executors-explained).
 
-## Configure the Kubernetes executor using pod templates
+## Configure the default worker Pod for all Deployments
 
-By default, Airflow Deployments on Astronomer use a pod template to construct each pod. To configure a Deployment's Kubernetes executor, you need to modify the Deployment's pod template and reapply a custom template via environment variables. To do so:
+By default, the Kubernetes executor launches workers based on a `podTemplate` configuration in the [Astronomer Airflow Helm chart](https://github.com/astronomer/airflow-chart/blob/master/values.yaml).
 
-1. Run the following command to find the namespace (release name) of your Airflow Deployment:
+You can modify the default `podTemplate` to configure the default worker Pods for all Deployments using the Kubernetes executor on your Astronomer Software installation.
```
Comment (jwitz) [inline software/kubernetes-executor.md:26]:
```suggestion
You can modify the default `podTemplate` to configure the default worker Pods for all Deployments using the Kubernetes executor on your Astronomer Software installation. You can then override this default at the task level using a `pod_override` file. See [Configure the worker Pod for a specific task](#configure-the-worker-pod-for-a-specific-task).
```

File: software/kubernetes-executor.md [hunk 13]
```diff
@@ -5,71 +5,189 @@ id: kubernetes-executor
...
 
-This enables the executor to scale depending on how many Airflow tasks you're running at a given time. It also means you can configure the following for each individual Airflow task:
+You can customize your Kubernetes Pods to scale depending on how many Airflow tasks you're running at a given time. It also means you can configure the following for each individual Airflow task:
 
 - Memory allocation
 - Service accounts
 - Airflow image
 
-To configure these resources for each Pod, you configure a Pod template. For more information on configuring Pod template values, reference the [Kubernetes documentation](https://kubernetes.io/docs/concepts/workloads/pods/#pod-templates).
+To configure these resources for a given tasks's Pod, you specify a `pod_override` in your DAG code. To specify a Pod template for many or all of your tasks, you can write a helper function to construct a `pod_override` in your DAGs or configure a global setting. For more information on configuring Pod template values, reference the [Kubernetes documentation](https://kubernetes.io/docs/concepts/workloads/pods/#pod-templates).
```
Comment (jwitz) [inline software/kubernetes-executor.md:16]:
```suggestion
To configure these resources for a given task's Pod, you specify a `pod_override` in your DAG code. To specify a Pod template for many or all of your tasks, you can write a helper function to construct a `pod_override` in your DAGs or configure a global setting. For more information on configuring Pod template values, reference the [Kubernetes documentation](https://kubernetes.io/docs/concepts/workloads/pods/#pod-templates).
```

**Pull request merged**